File: experiment-14/runs/language=typescript_model=opus-4.8-fast_prompt=TDD/rep1/tests/format.test.ts
import { describe, it, expect } from "vitest";
import {
  formatMatch,
  formatMatchList,
  formatHeadToHead,
  formatTeamRecord,
  formatStandings,
  formatStatistics,
  formatPlayer,
  formatPlayerList,
} from "../src/format.js";
import type { Match, Player, TeamRecord } from "../src/types.js";
import { normalizeTeamName, normalizeName, parseDate } from "../src/normalize.js";

const match: Match = {
  competition: "Brasileirão Série A",
  date: parseDate("2023-09-03"),
  season: 2023,
  round: "22",
  homeTeam: "Flamengo-RJ",
  awayTeam: "Fluminense-RJ",
  homeKey: "flamengo",
  awayKey: "fluminense",
  homeGoals: 2,
  awayGoals: 1,
  source: "test",
};

describe("formatMatch", () => {
  it("renders date, score and competition context", () => {
    expect(formatMatch(match)).toBe(
      "2023-09-03: Flamengo-RJ 2-1 Fluminense-RJ (Brasileirão Série A Round 22)"
    );
  });

  it("includes stage when present and omits round", () => {
    const liber: Match = {
      ...match,
      competition: "Copa Libertadores",
      round: undefined,
      stage: "final",
    };
    expect(formatMatch(liber)).toBe(
      "2023-09-03: Flamengo-RJ 2-1 Fluminense-RJ (Copa Libertadores, final)"
    );
  });
});

describe("formatMatchList", () => {
  it("renders a header and bullet list", () => {
    const out = formatMatchList([match], "Flamengo vs Fluminense");
    expect(out).toContain("Flamengo vs Fluminense");
    expect(out).toContain("- 2023-09-03:");
  });

  it("reports when no matches are found", () => {
    expect(formatMatchList([], "Nothing")).toContain("No matches found");
  });
});

describe("formatHeadToHead", () => {
  it("summarises wins and draws", () => {
    const out = formatHeadToHead({
      teamA: "Flamengo",
      teamB: "Fluminense",
      matches: 3,
      teamAWins: 2,
      teamBWins: 0,
      draws: 1,
      games: [],
    });
    expect(out).toContain("Flamengo 2 wins");
    expect(out).toContain("Fluminense 0 wins");
    expect(out).toContain("1 draw");
  });
});

describe("formatTeamRecord", () => {
  it("renders win/draw/loss, goals and win rate", () => {
    const rec: TeamRecord = {
      team: "Corinthians",
      matches: 19,
      wins: 11,
      draws: 5,
      losses: 3,
      goalsFor: 28,
      goalsAgainst: 15,
      points: 38,
    };
    const out = formatTeamRecord(rec, "Corinthians record");
    expect(out).toContain("Matches: 19");
    expect(out).toContain("Wins: 11, Draws: 5, Losses: 3");
    expect(out).toContain("Goals For: 28, Goals Against: 15");
    expect(out).toContain("57.9%");
  });
});

describe("formatStandings", () => {
  it("renders a numbered points table", () => {
    const table: TeamRecord[] = [
      { team: "Flamengo", matches: 38, wins: 28, draws: 6, losses: 4, goalsFor: 86, goalsAgainst: 37, points: 90 },
      { team: "Santos", matches: 38, wins: 22, draws: 8, losses: 8, goalsFor: 60, goalsAgainst: 40, points: 74 },
    ];
    const out = formatStandings(table, "2019 Brasileirão");
    expect(out).toContain("1. Flamengo - 90 pts");
    expect(out).toContain("2. Santos - 74 pts");
  });
});

describe("formatStatistics", () => {
  it("renders averages and rates", () => {
    const out = formatStatistics(
      {
        totalMatches: 100,
        totalGoals: 247,
        averageGoals: 2.47,
        homeWins: 47,
        awayWins: 30,
        draws: 23,
        homeWinRate: 0.473,
        awayWinRate: 0.3,
        drawRate: 0.227,
      },
      "Brasileirão"
    );
    expect(out).toContain("Average goals per match: 2.47");
    expect(out).toContain("Home win rate: 47.3%");
  });
});

describe("formatPlayer / formatPlayerList", () => {
  const player: Player = {
    id: 1,
    name: "Neymar Jr",
    nameKey: normalizeName("Neymar Jr"),
    age: 31,
    nationality: "Brazil",
    overall: 92,
    potential: 92,
    club: "Paris Saint-Germain",
    clubKey: normalizeTeamName("Paris Saint-Germain"),
    position: "LW",
    jerseyNumber: 10,
    height: "5'9",
    weight: "150lbs",
  };

  it("formats a single player line", () => {
    expect(formatPlayer(player)).toBe(
      "Neymar Jr - Overall: 92, Position: LW, Club: Paris Saint-Germain"
    );
  });

  it("numbers a player list under a header", () => {
    const out = formatPlayerList([player], "Top players");
    expect(out).toContain("Top players");
    expect(out).toContain("1. Neymar Jr - Overall: 92");
  });

  it("reports when no players are found", () => {
    expect(formatPlayerList([], "x")).toContain("No players found");
  });
});

File: experiments/adrianco/experiment-15-sonnet5/brazil/runs/language=go_model=sonnet-5_prompt=tdd/rep1/internal/mcpserver/format_test.go
package mcpserver

import (
	"strings"
	"testing"

	"brazilian-soccer-mcp/internal/soccer"
)

func mustDate(t *testing.T, s string) soccer.Match {
	t.Helper()
	d, err := soccer.ParseDate(s)
	if err != nil {
		t.Fatal(err)
	}
	return soccer.Match{Date: d}
}

func TestFormatMatch(t *testing.T) {
	m := soccer.Match{
		Competition: "Brasileirao",
		Round:       "22",
		HomeTeam:    "Flamengo",
		AwayTeam:    "Fluminense",
		HomeGoals:   2,
		AwayGoals:   1,
	}
	m.Date = mustDate(t, "2023-09-03").Date

	got := FormatMatch(m)
	want := "2023-09-03: Flamengo 2-1 Fluminense (Brasileirao Round 22)"
	if got != want {
		t.Errorf("FormatMatch() = %q, want %q", got, want)
	}
}

func TestFormatMatchNoRound(t *testing.T) {
	m := soccer.Match{
		Competition: "Libertadores",
		HomeTeam:    "River Plate",
		AwayTeam:    "Boca Juniors",
		HomeGoals:   3,
		AwayGoals:   1,
	}
	m.Date = mustDate(t, "2018-12-08").Date

	got := FormatMatch(m)
	want := "2018-12-08: River Plate 3-1 Boca Juniors (Libertadores)"
	if got != want {
		t.Errorf("FormatMatch() = %q, want %q", got, want)
	}
}

func TestFormatMatchesEmpty(t *testing.T) {
	got := FormatMatches(nil)
	if !strings.Contains(got, "No matches found") {
		t.Errorf("FormatMatches(nil) = %q, want a no-matches message", got)
	}
}

func TestFormatHeadToHead(t *testing.T) {
	r := soccer.HeadToHeadResult{
		TeamA:     "Flamengo",
		TeamB:     "Fluminense",
		TeamAWins: 1,
		TeamBWins: 1,
		Draws:     1,
	}
	got := FormatHeadToHead(r)
	if !strings.Contains(got, "Flamengo 1 wins") || !strings.Contains(got, "Fluminense 1 wins") || !strings.Contains(got, "1 draws") {
		t.Errorf("FormatHeadToHead() = %q, missing expected record summary", got)
	}
}

func TestFormatTeamRecord(t *testing.T) {
	r := soccer.TeamRecord{
		Team: "Corinthians", Matches: 19, Wins: 11, Draws: 5, Losses: 3,
		GoalsFor: 28, GoalsAgainst: 15, WinRate: 57.9,
	}
	got := FormatTeamRecord(r)
	if !strings.Contains(got, "Corinthians") || !strings.Contains(got, "Wins: 11") || !strings.Contains(got, "57.9%") {
		t.Errorf("FormatTeamRecord() = %q, missing expected fields", got)
	}
}

func TestFormatStandings(t *testing.T) {
	table := []soccer.StandingEntry{
		{Team: "Flamengo", Played: 38, Wins: 28, Draws: 6, Losses: 4, GoalsFor: 88, GoalsAgainst: 40, GoalDiff: 48, Points: 90},
	}
	got := FormatStandings(table, "Brasileirao", 2019)
	if !strings.Contains(got, "1. Flamengo - 90 pts") {
		t.Errorf("FormatStandings() = %q, missing expected standing line", got)
	}
}

func TestFormatStatsSummary(t *testing.T) {
	sum := soccer.Summary{TotalMatches: 100, AvgGoalsPerMatch: 2.47, HomeWinRate: 47.3, AwayWinRate: 30.0, DrawRate: 22.7}
	got := FormatStatsSummary(sum)
	if !strings.Contains(got, "2.47") || !strings.Contains(got, "47.3%") {
		t.Errorf("FormatStatsSummary() = %q, missing expected fields", got)
	}
}

func TestFormatPlayers(t *testing.T) {
	players := []soccer.Player{
		{Name: "Neymar Jr", Overall: 92, Position: "LW", Club: "Paris Saint-Germain"},
	}
	got := FormatPlayers(players)
	if !strings.Contains(got, "1. Neymar Jr - Overall: 92, Position: LW, Club: Paris Saint-Germain") {
		t.Errorf("FormatPlayers() = %q, missing expected player line", got)
	}
}

func TestFormatPlayersEmpty(t *testing.T) {
	got := FormatPlayers(nil)
	if !strings.Contains(got, "No players found") {
		t.Errorf("FormatPlayers(nil) = %q, want a no-players message", got)
	}
}
